File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/watchproviders/model/WatchProviders.java
package dev.reuss.tmdb.domain.watchproviders.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * TMDB watch provider list.
 *
 * @param results watch providers
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record WatchProviders(
        List<WatchProvider> results
) implements TmdbModel {

    public WatchProviders {
        results = TmdbCollections.list(results);
    }

}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/watchproviders/model/WatchProviders.kt
package dev.reuss.tmdb.domain.watchproviders.model

import com.fasterxml.jackson.annotation.JsonProperty

import dev.reuss.tmdb.common.TmdbModel

/**
 * TMDB watch provider list.
 *
 * @property results watch providers
 */
@JvmRecord
data class WatchProviders(
    @all:JsonProperty("results")
    val results: List<WatchProvider> = emptyList()
) : TmdbModel
